src/main/java/dev/hephaestus/glowcase/networking/packet/EditItemDisplayBlockSettings.java

```diff
@@ -0,0 +1,72 @@
+import dev.hephaestus.glowcase.block.entity.ItemDisplayBlockEntity;
+import dev.hephaestus.glowcase.util.NetworkingUtil;
+import net.fabricmc.fabric.api.client.networking.v1.ClientPlayNetworking;
+import net.fabricmc.fabric.api.networking.v1.ServerPlayNetworking;
+import net.minecraft.network.RegistryByteBuf;
+import net.minecraft.network.codec.PacketCodec;
+import net.minecraft.network.codec.PacketCodecs;
+import net.minecraft.network.packet.CustomPayload;
+import net.minecraft.server.world.ServerWorld;
+import net.minecraft.state.property.Properties;
+import net.minecraft.util.math.BlockPos;
+import net.minecraft.util.math.RotationPropertyHelper;
+
+public record EditItemDisplayBlockSettings(BlockPos pos, ItemDisplayBlockEntity.RotationType rotationType,
+                                           ItemDisplayBlockEntity.GivesItem givesItem,
+                                           ItemDisplayBlockEntity.Offset offset,
+                                           ItemDisplayBlockValues values) implements CustomPayload {
+
+	public static final PacketCodec<RegistryByteBuf, EditItemDisplayBlockSettings> PACKET_CODEC = PacketCodec.tuple(
+		BlockPos.PACKET_CODEC, EditItemDisplayBlockSettings::pos,
+		PacketCodecs.BYTE.xmap(index -> ItemDisplayBlockEntity.RotationType.values()[index], rotation -> (byte) rotation.ordinal()), EditItemDisplayBlockSettings::rotationType,
+		PacketCodecs.BYTE.xmap(index -> ItemDisplayBlockEntity.GivesItem.values()[index], givesItem -> (byte) givesItem.ordinal()), EditItemDisplayBlockSettings::givesItem,
+		PacketCodecs.BYTE.xmap(index -> ItemDisplayBlockEntity.Offset.values()[index], offset -> (byte) offset.ordinal()), EditItemDisplayBlockSettings::offset,
+		ItemDisplayBlockValues.PACKET_CODEC, EditItemDisplayBlockSettings::values,
+		EditItemDisplayBlockSettings::new
+	);
+
+	public static final Id<EditItemDisplayBlockSettings> PACKET_ID = new Id<>(Glowcase.id("channel.item_display"));
+
+	@Override
+	public Id<? extends CustomPayload> getId() {
+		return PACKET_ID;
+	}
+
+	public void send() {
+		ClientPlayNetworking.send(this);
+	}
+
+	public void receive(ServerPlayNetworking.Context context) {
+		ServerWorld serverWorld = context.player().getServerWorld();
+		if (this.values().rotation() < 0 || this.values().rotation() >= RotationPropertyHelper.getMax()) return;
+		if (!NetworkingUtil.canEditGlowcase(context.player(), this.pos(), Glowcase.ITEM_DISPLAY_BLOCK.get())) return;
+		if (!(serverWorld.getBlockEntity(this.pos()) instanceof ItemDisplayBlockEntity be)) return;
+
+		be.givesItem = this.givesItem();
+		be.rotationType = this.rotationType();
+		be.offset = this.offset();
+		be.pitch = this.values().pitch();
+		be.yaw = this.values().yaw();
+		be.showName = this.values().showName();
+
+		serverWorld.setBlockState(this.pos(), context.player().getWorld()
+			.getBlockState(this.pos()).with(Properties.ROTATION, this.values().rotation()));
+
+		be.markDirty();
+		be.dispatch();
+	}
+
+	// separated for tuple call
```

Comment:
is this just over the limit of PacketCodec.tuple?

Reply:
`.tuple(...)` is only implemented for max. 6 parameters, and the record would've needed ~9.
So if there are more, it can be packed into an extra `PACKET_CODEC` to nest it.

Also, arguably, is a bit more readable too. ^^

Reply:
ah yeah and I see it was already like that originally - cheers for the clarification!